test/jdk/java/beans/Introspector/DefaultMethodBeanPropertyTest.java

```diff
@@ -64,30 +77,135 @@ public Float getObj() {
+        verifyProperties(D2.class,
+            "getClass",
+            "getFoo"
+        );
+    }
+
+//////////////////////////////////////
+//                                  //
+//          SCENARIO 3              //
+//                                  //
+//////////////////////////////////////
+
+    public interface A3 {
+        default Object getFoo() {
+            return null;
+        }
+    }
+
+    public interface B3 extends A3 {
+        @Override
+        Set<?> getFoo();
+    }
+
+    public interface C3 extends A3 {
+        @Override
+        Collection<?> getFoo();
+    }
+
+    public class D3 implements B3, C3 {
+        @Override
+        public NavigableSet<?> getFoo() {
+            return null;
+        }
+    }
+
+    public static void testScenario3() {
+        verifyProperties(D3.class,
+            "getClass",
+            "getFoo"
+        );
+    }
+
+// Helper methods
+
+    public static void verifyProperties(Class<?> type, String... getterNames) {
+
+        // Gather expected properties
         final HashSet<PropertyDescriptor> expected = new HashSet<>();
-        expected.add(new PropertyDescriptor("class", ClassB.class, "getClass", null));  // inherited method
-        expected.add(new PropertyDescriptor("value", ClassB.class, "getValue", null));  // inherited default method
-        expected.add(new PropertyDescriptor("foo", ClassB.class, "getFoo", null));      // overridden interface method
-        expected.add(new PropertyDescriptor("obj", ClassB.class, "getObj", null));      // overridden default method
+        for (String methodName : getterNames) {
+            final String suffix = methodName.substring(3);
+            final String propName = Introspector.decapitalize(suffix);
+            final Method getter;
+            try {
+                getter = type.getMethod(methodName);
+            } catch (NoSuchMethodException e) {
+                throw new Error("unexpected error", e);
+            }
+            final PropertyDescriptor propDesc;
+            try {
+                propDesc = new PropertyDescriptor(propName, getter, null);
+            } catch (IntrospectionException e) {
+                throw new Error("unexpected error", e);
+            }
+            expected.add(propDesc);
+        }
+
+        // Verify properties can be found directly
+        expected.stream()
+          .map(PropertyDescriptor::getName)
+          .filter(name -> BeanUtils.getPropertyDescriptor(type, name) == null)
+          .findFirst()
+          .ifPresent(name -> {
+            throw new Error("property \"" + name + "\" not found in " + type);
+          });
 
-        // Actual properties
-        final HashSet<PropertyDescriptor> actual = new HashSet<>(
-          Arrays.asList(BeanUtils.getPropertyDescriptors(ClassB.class)));
+        // Gather actual properties
+        final Set<PropertyDescriptor> actual = Set.of(
+            BeanUtils.getPropertyDescriptors(type));
```

Comment:
```suggestion
        final Set<PropertyDescriptor> actual =
                Set.of(BeanUtils.getPropertyDescriptors(type));
```
May be easier to read? I mean wrapping the entire expression instead of breaking the expression.

Reply:
Thanks - fixed in b92726a923b.